### PR TITLE
fix: Resolve menu.py not found error when running `pip install deadline-cloud-for-nuke -t <folder>`

### PR DESCRIPTION
**Pull request by Cherie-Chen**
### What was the problem/requirement? (What/Why)

Addressing user feedback on receiving `<folder> doesn't have menu.py` error while installing `deadline-cloud-for-nuke` using `pip install deadline-cloud-for-nuke -t <folder>` command in MacOS

### What was the solution? (How)

Move `menu.py` to the correct folder

### What is the impact of this change?

Better customer experience in installing `deadline-cloud-for-nuke` 

### How was this change tested?

```
% ./python -m pip install -e /Users/cheriech/workplace/deadline-cloud-for-nuke -t ~/playground/nuke_installation_test
...
Successfully built deadline-cloud-for-nuke
Installing collected packages: xxhash, urllib3, typing-extensions, six, pyyaml, pyrsistent, packaging, jmespath, click, attrs, qtpy, python-dateutil, jsonschema, openjd-adaptor-runtime, botocore, s3transfer, boto3, deadline, deadline-cloud-for-nuke
Successfully installed attrs-23.2.0 boto3-1.34.147 botocore-1.34.147 click-8.1.7 deadline-0.48.7 deadline-cloud-for-nuke-0.18.3.post3+g08b2b27.d20240724 jmespath-1.0.1 jsonschema-4.17.3 openjd-adaptor-runtime-0.8.0 packaging-24.1 pyrsistent-0.20.0 python-dateutil-2.9.0.post0 pyyaml-6.0.1 qtpy-2.4.1 s3transfer-0.10.2 six-1.16.0 typing-extensions-4.12.2 urllib3-2.2.2 xxhash-3.4.1
```

### Did you run the "Job Bundle Output Tests"? If not, why not? If so, paste the test results here.

```
Timestamp: 2024-07-24T09:49:01.895634-07:00
Running job bundle output test: /Users/cheriech/workplace/deadline-cloud-for-nuke/job_bundle_output_tests/group-read

group-read
Test succeeded

Timestamp: 2024-07-24T09:49:02.561327-07:00
Running job bundle output test: /Users/cheriech/workplace/deadline-cloud-for-nuke/job_bundle_output_tests/noise-saver

noise-saver
Test succeeded

Timestamp: 2024-07-24T09:49:03.142422-07:00
Running job bundle output test: /Users/cheriech/workplace/deadline-cloud-for-nuke/job_bundle_output_tests/multi-load-save

multi-load-save
Test succeeded

Timestamp: 2024-07-24T09:49:03.698305-07:00
Running job bundle output test: /Users/cheriech/workplace/deadline-cloud-for-nuke/job_bundle_output_tests/ocio

ocio
Test succeeded

Timestamp: 2024-07-24T09:49:04.180436-07:00
Running job bundle output test: /Users/cheriech/workplace/deadline-cloud-for-nuke/job_bundle_output_tests/cwd-path

cwd-path
Test succeeded

All tests passed, ran 5 total.
Timestamp: 2024-07-24T09:49:09.928971-07:00
```

### Was this change documented?

Yes, `DEVELOPMENT.md` file is updated with new `menu.py` path.

### Is this a breaking change?

No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: install_builder/deadline-cloud-for-nuke.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
                     <onPackingFilterList>
                         <fileNameFilter pattern="*/deadline/nuke_submitter/*" logic="matches" patternType="glob"/>
                         <fileNameFilter pattern="*/deadline/nuke_util/*" logic="matches" patternType="glob"/>
-                        <fileNameFilter pattern="*/menu.py" logic="matches" patternType="glob"/>
+                        <fileNameFilter pattern="*/deadline/nuke_submitter/menu.py" logic="matches" patternType="glob"/>
```
Comment (epmog) [inline install_builder/deadline-cloud-for-nuke.xml]: nit: Line 19 now seems to have the menu.py covered

File: src/deadline/nuke_submitter/menu.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            "Submit to Deadline Cloud", show_nuke_render_submitter_noargs, ""
+        )
+        # Set the environment variable DEADLINE_ENABLE_DEVELOPER_OPTIONS to "true" to get this menu.
+        if os.environ.get("DEADLINE_ENABLE_DEVELOPER_OPTIONS", "").upper() == "TRUE":
+            aws_deadline_menu.addCommand(
+                "Run Nuke Submitter Job Bundle Output Tests...",
+                run_render_submitter_job_bundle_output_test,
+                "",
+            )
+    except BaseException:
```
Comment (epmog) [inline src/deadline/nuke_submitter/menu.py]: SonarCloud is flagging this as an issue. I know you just moved it but we can at least shrink this down to `Exception`. Shouldn't have a need for `BaseException`.

Can see it here or from the bot comment: https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=157&id=aws-deadline_deadline-cloud-for-nuke&open=AZDmKG3AQEoF-EVdt0fJ&tab=why

File: README.md [hunk 3]
```diff
@@ -33,6 +33,14 @@ The Nuke Adaptor implements the [OpenJD][openjd-adaptor-runtime] interface that
 
 Jobs created by the submitter use this adaptor by default.
 
+### Compatibility
+
+This library requires:
+
+1. Nuke 15,
+1. Python 3.9 or higher; and
+1. Linux operating system.
```
Comment (epmog) [inline README.md]: The adaptor should work on macos, linux, and windows? Although maybe Evan's PR fixes Windows, i'd have to double check. This _should_ be the same compatibility as the submitter currently. Let me know if I'm missing something

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -53,7 +53,7 @@ WARNING: This workflow installs additional Python packages into your Nuke's pyth
 2. Switch to your Nuke directory, like `cd "C:\Program Files\Nuke15.0v2"`.
 3. Run `.\python -m pip install -e C:\Users\<username>\deadline-clients\deadline-cloud` to install the AWS Deadline Cloud Client Library in edit mode.
 4. Run `.\python -m pip install -e C:\Users\<username>\deadline-clients\deadline-cloud-for-nuke` to install the Nuke Submitter in edit mode.
-5. Run `set NUKE_PATH=C:\Users\<username>\deadline-clients\deadline-cloud-for-nuke\src` to put the `menu.py` file in the path Nuke searches for menu extensions.
+5. Run `set NUKE_PATH=C:\Users\<username>\deadline-clients\deadline-cloud-for-nuke\src\deadline\nuke_submitter` to put the `menu.py` file in the path Nuke searches for menu extensions.
```
Comment (epmog) [inline DEVELOPMENT.md:56]: Part of the reason why we wanted to include menu.py in the built wheel in the first place was for users to be able to install the integration using pip (i.e., a released version)

Now that it's in the top-level of the nuke submitter, it should also be easy to provide instructions on how users can do that workflow. 

Can we add that in this change as well?

**Review by epmog — CHANGES_REQUESTED** [review]
🎉 Congrats on your first PR Cherie! 🎉 This is looking good so I just have a few comments.

I believe there's a missing change in the installer that will prevent the submitter installer from working. Now that we've changed the location of the menu.py, the `NUKE_PATH` environment variable will also have to be updated to the new location.

File: install_builder/deadline-cloud-for-nuke.xml (hunk 1 above)
Comment (Cherie-Chen) [reply install_builder/deadline-cloud-for-nuke.xml]: Make sense. Will remove this in the next revision.

File: src/deadline/nuke_submitter/menu.py (hunk 2 above)
Comment (Cherie-Chen) [reply src/deadline/nuke_submitter/menu.py]: Sounds good. I will change this in the next revision.

File: README.md (hunk 3 above)
Comment (Cherie-Chen) [reply README.md]: I see. I will revert the changes I have in README. Thanks!

**Comment by epmog** [comment]
https://github.com/aws-deadline/deadline-cloud-for-nuke/pull/165 was merged that takes in these changes and addresses the last open comments. Thanks Cherie for your contribution! Closing this PR.

**Pull request closed without merging**